PYME/Acquire/Hardware/pco/pco_cam.py

```diff
@@ -68,25 +73,21 @@ def CamReady(self):
         return self.initalized
 
     def ExtractColor(self, chSlice, mode):
-
         # Somehow this check matters... shouldn't this be taken care of by ExpReady???
-        if self.GetNumImsBuffered() < 1:
-            return True
+        if (not self.recording) or (self.GetNumImsBuffered() < 1):
+            raise RuntimeError('Trying to grab a frame when there is no frame to grab.')
 
         if self.n_read < self.buffer_size:
             curr_frame = self.n_read
         else:
             curr_frame = self.buffer_size-self.GetNumImsBuffered()
```

Comment:
This logic looks wrong for a circular buffer ... shouldn't it be something like `n_read % self.buffer_size`

Reply:
It's a ring buffer, not a circular buffer, so the position of the latest frame moves as data populates. I promise this code is extensively tested & the logic works minus the one strange bug I mentioned already. Most of the time, this camera runs fine.